src/core/wal.rs

```diff
@@ -58,7 +58,7 @@ pub struct Wal {
 }
 
 impl Wal {
-    pub fn new(dir: &str) -> Result<Wal, io::Error> {
+    pub fn create(dir: &str) -> Result<Wal, io::Error> {
```

Comment:
[`clippy` said: method `new` should return `Self`.](https://rust-lang.github.io/rust-clippy/master/index.html#new_ret_no_self)